graphql/exec.go

```diff
@@ -21,12 +21,66 @@ func CollectFields(doc *query.Document, selSet []query.Selection, satisfies []st
 	return collectFields(doc, selSet, satisfies, variables, map[string]bool{})
 }
 
+func runDirectives(directives common.DirectiveList, variables map[string]interface{}) bool {
+	for _, directive := range directives {
+		if !runSingleDirective(directive, variables) {
+			return false
+		}
+	}
+	return true
+}
+
+func runSingleDirective(directive *common.Directive, variables map[string]interface{}) bool {
+	arg := directive.Args.MustGet("if")
+	fieldAllowed := true
+	switch arg.(type) {
+	case *common.Variable:
+		value := arg.Value(variables)
```

Comment:
Why can't you pass the operation in here and do the variable lookup only when its actually referenced, instead of every resolver?

`populateVariables` is a big smell to me, its going to run on every resolver regardless of if there is even a directive.

Reply:
Will fix ASAP!